### ISSUE TITLE
Get field error code

### ISSUE THREAD

**Issue by mohic** [thymeleaf/thymeleaf-spring#90]
Hi,

In my project, I want to get error code from a specific field but this method is not available.

For the moment I have resolved my problem with this code:

```
<th:block th:each="detailedError : ${#fields.detailedErrors()}">
    <span th:if="${detailedError.getFieldName() == 'emailAddress'} and ${detailedError.getCode() == 'myProject.subscription.validation.emailAddress.alreadyExists'}" class="textError">This e-mail address is already exists</span>
</th:block>
```

It will be great if you add possibility to make:

```
${#fields.errorCode('emailAddress')}
```

Best Regards,
Christophe

**Comment by gauthierj** [comment]
+1 on this one, would be nice to have this.

This seems to be more a bug than a feature. In FieldUtils.computeErrors(...) (around line 139) the code :

final String[] errorCodes = bindStatus.getErrorMessages();

seems to indicate that error codes and not error messages are supposed to  be returned.

Regards

**Comment by saljuama** [comment]
+1, is there any better place to vote for that feature request?

**Comment by danielfernandez** [comment]
This has been implemented for Thymeleaf 3.0. Instead of creating an `errorCode(...)` method in `#fields`, the `detailedErrors(...)` functionality has been expanded so that users can now ask not only for `DetailedError` objects for the entire form, but also for a specific field or only those that are marked _global_ (i.e. not belonging to any specific field).

The new methods mimic the method names for `errors(...)`, and are:

```
#fields.allDetailedErrors()
#fields.detailedErrors()
#fields.detailedErrors(fieldName)
#fields.globalDetailedErrors()
```

All these methods return `List<DetailedError>`. All `DetailedError` objects contain the error field, code, message, type and arguments.

**Comment by danielfernandez** [comment]
Already in `3.0.0-SNAPSHOT`
